noria/src/data.rs

```diff
@@ -516,13 +516,20 @@ macro_rules! arithmetic_operation (
         match ($first, $second) {
             (&DataType::None, _) | (_, &DataType::None) => DataType::None,
             (&DataType::Int(a), &DataType::Int(b)) => (a $op b).into(),
+            (&DataType::UnsignedInt(a), &DataType::UnsignedInt(b)) => (a $op b).into(),
             (&DataType::BigInt(a), &DataType::BigInt(b)) => (a $op b).into(),
+            (&DataType::UnsignedBigInt(a), &DataType::UnsignedBigInt(b)) => (a $op b).into(),
             (&DataType::Int(a), &DataType::BigInt(b)) => (i64::from(a) $op b).into(),
+            (&DataType::Int(a), &DataType::UnsignedBigInt(b)) => (i128::from(a) $op i128::from(b)).into(),
             (&DataType::BigInt(a), &DataType::Int(b)) => (a $op i64::from(b)).into(),
-            // TODO ADD SUPPORT FOR MATHS ON SIGNED (Big)Int
+            (&DataType::UnsignedBigInt(a), &DataType::Int(b)) => (i128::from(a) $op i128::from(b)).into(),
+            (&DataType::UnsignedBigInt(a), &DataType::UnsignedInt(b)) => (a $op u64::from(b)).into(),
+            (&DataType::UnsignedInt(a), &DataType::UnsignedBigInt(b)) => (u64::from(a) $op b).into(),
 
             (first @ &DataType::Int(..), second @ &DataType::Real(..)) |
+            (first @ &DataType::UnsignedInt(..), second @ &DataType::Real(..)) |
             (first @ &DataType::Real(..), second @ &DataType::Int(..)) |
+            (first @ &DataType::Real(..), second @ &DataType::UnsignedInt(..)) |
```

Comment:
No (`Unsigned`)`BigInt` here?

Reply:
The max and min values of `f64` (64-bit floating point) fit inside the `i128` range, so comparison with `BigInt` seems fine. Do you have a reason to think we don't want it here?

Reply:
That's why I was surprised (Signed)`BigInt` wasn't there. I'll add it.